tinyDA/diagnostics.py

```diff
@@ -207,3 +211,58 @@ def get_samples(chain, attribute="parameters", level="fine", burnin=0):
 
     # return the samples.
     return samples
+
+
+def DA_estimator(chain, attribute="qoi", variable="x0", burnin=0):
+    """Computes the unbiased Monte-Carlo estimator for Delayed Acceptance (two-level),
+    as derived in Lykkegaard et al. 2023.
+
+    Parameters
+    ----------
+    chain : dict
+        A dict as returned by tinyDA.sample, containing chain information
+        and lists of tinyDA.Link instances.
+    attribute : str, optional
+        Which link attribute ('parameters', 'model_output', 'qoi' or 'stats')
+        to extract. The default is 'parameters'.
+    variable : str, optional
+        Which variable of the posterior or qoi to marginalize over.
+    burnin : int, optional
+        The burnin length. The default is 0.
+
+    Returns
+    ----------
+    float
+        Outpu of the estimator computation.
+
+    """
+
+    subchain_length = chain['subchain_length']
+
+    inferencedata_coarse = to_inference_data(chain, level="coarse", burnin=burnin*subchain_length)
+    inferencedata_coarse_promoted = to_inference_data(
+        chain, level="promoted_coarse", burnin=burnin
+    )
+    inferencedata_fine = to_inference_data(chain, level="fine", burnin=burnin)
+
+    values_coarse_promoted = 0
+    values_fine = 0
+    values_coarse = 0
+
+    if attribute == "qoi":
+        values_coarse_promoted = inferencedata_coarse_promoted.qoi
+        values_fine = inferencedata_fine.qoi
+        values_coarse = inferencedata_coarse.qoi
+    if attribute == "posterior":
+        values_coarse_promoted = inferencedata_coarse_promoted.posterior
+        values_fine = inferencedata_fine.posterior
+        values_coarse = inferencedata_coarse.posterior
+
+    values_difference = values_coarse_promoted - values_fine
+
+    mean_difference = az.summary(values_difference, var_names=[variable]).loc[
+        variable, "mean"
+    ]
+    mean_coarse = az.summary(values_coarse, var_names=[variable]).loc[variable, "mean"]
```

Comment:
Why not just compute the mean directly rather than the entire summary and then extract the mean. There should be a faster way of doing this.